src/attr/_make.py

```diff
@@ -181,22 +181,24 @@ class MyClassAttributes(tuple):
     return globs[attr_class_name]
 
 
+_Attributes = _make_attr_tuple_class("_Attributes", [
+    "attrs", "super_attrs", "counting_attrs",
+])
+
+
 def _transform_attrs(cls, these):
     """
-    Transform all `_CountingAttr`s on a class into `Attribute`s and save the
-    list in `__attrs_attrs__` while potentially deleting them from *cls*.
+    Transform all `_CountingAttr`s on a class into `Attribute`s.
 
     If *these* is passed, use that and don't look for them on the class.
 
-    Return a list of tuples of (attribute name, attribute).
+    Return an `_Attributes`.
     """
     if these is None:
         ca_list = [(name, attr)
                    for name, attr
                    in cls.__dict__.items()
                    if isinstance(attr, _CountingAttr)]
-        for name, _ in ca_list:
-            delattr(cls, name)
```

Comment:
To check my understanding: This was only needed because the old class was still hanging around, and now it's not because they're just not copied to the newly created one anyway class?

Reply:
Yeah, the original approach was to stick everything on the old class and if it’s a slots class, copy the result to a new one.